### ISSUE TITLE
False positive clippy::needless_lifetimes in async fn taking two parameters by reference

### ISSUE THREAD

**Issue by Hawk777** [rust-lang/rust#86576]
Given [the following code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=43453df53763dd40babe88f1ea11fb5c):

```rust
#![warn(clippy::needless_lifetimes)]

pub async fn do_thing<'a>(s: &'a str, s2: &str) -> &'a str {
	println!("{}", s2);
	s
}

fn main() {
}
```

The Clippy output is:

```
warning: explicit lifetimes given in parameter types where they could be elided (or replaced with `'_` if needed by type declaration)
 --> src/main.rs:3:1
  |
3 | pub async fn do_thing<'a>(s: &'a str, s2: &str) -> &'a str {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
```

This warning is a false positive. The `'a` lifetime cannot be elided because there are two reference inputs; removing all lifetimes, as expected, results in a compile error (“the signature does not say whether it is borrowed from `s` or `s2`”). Additionally, removing `async` from the function makes the warning go away, so this is `async`-specific.

**Comment by smklein** [comment]
FYI this might get more attention filed in Clippy's repo: https://github.com/rust-lang/rust-clippy

**Comment by jonas-schievink** [comment]
duplicate of https://github.com/rust-lang/rust-clippy/issues/7271
